shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/config/datasource/DataSourceConfiguration.java

```diff
@@ -111,9 +112,13 @@ public DataSource createDataSource() {
             if (SKIPPED_PROPERTY_NAMES.contains(entry.getKey())) {
                 continue;
             }
-            Optional<Method> setterMethod = findSetterMethod(methods, entry.getKey());
-            if (setterMethod.isPresent()) {
-                setterMethod.get().invoke(result, entry.getValue());
+            try {
+                Optional<Method> setterMethod = findSetterMethod(methods, entry.getKey());
+                if (setterMethod.isPresent()) {
+                    setterMethod.get().invoke(result, entry.getValue());
+                }
+            } catch (Exception ex) {
+                throw new ShardingSphereConfigurationException(ex, String.format("datasource property %s configure error", entry.getKey()));
```

Comment:
The error info needs more consideration since that will be exposed to users.
How about `Incorrect configuration item: the property %s of the dataSource`?
That's just my draft. It's up to you, however please be careful of English grammar.

Reply:
how about this exception message: ·`String.format("Incorrect configuration item: the property %s of the dataSource. the reason is %s", ex.getExceptionMessage())`